knimin/lib/data_access.py

```diff
@@ -2588,110 +2609,172 @@ def _sample_exists(self, sample_id):
         bool
             Whether the sample exists
         """
-        sql = """SELECT EXISTS (SELECT 1 FROM pm.sample
-                                WHERE sample_id = %s)"""
-        return self._con.execute_fetchone(sql, [sample_id])[0]
+        with TRN:
+            sql = """SELECT EXISTS (SELECT 1 FROM pm.sample
+                                    WHERE sample_id = %s)"""
+            TRN.add(sql, [sample_id])
+            return TRN.execute_fetchlast()
 
-    def _barcode_exists(self, barcode):
-        """Confirms that a barcode exists
+    def _samples_exist(self, sample_ids, exist=True):
+        """Confirms that given sample ID(s) (do not) exist
 
         Parameters
         ----------
-        barcode : str
-            Barcode to check
+        sample_ids : list of str
+            IDs of the sample to check
+        exist : bool, optional, default: True
+            True: checks if exist
+            False: checks if not exist
 
         Raises
         ------
         ValueError
-            If the barcode does not exist
+            If one or more sample IDs do not exist (when exist == True)
+            If one or more sample IDs already exist (when exist == False)
         """
-        sql = """SELECT EXISTS (SELECT 1 FROM barcodes.barcode
-                                WHERE barcode = %s)"""
-        if not self._con.execute_fetchone(sql, [barcode])[0]:
-            raise ValueError('Barcode %s does not exist.' % barcode)
+        with TRN:
+            sql = """SELECT sample_id
+                     FROM pm.sample
+                     WHERE sample_id IN %s"""
+            TRN.add(sql, [tuple(sample_ids)])
+            res = [x[0] for x in TRN.execute_fetchindex()]
```

Comment:
I'm unsure, but I think `res = TRN.execute_fetchflatten()` will given you the same result in this call.